source/shared/core_conn.cpp

```diff
@@ -1005,6 +1066,48 @@ void column_encryption_set_func::func( _In_ connection_option const* option, _In
     conn_str += ";";
 }
 
+void ce_akv_str_set_func::func( _In_ connection_option const* option, _In_ zval* value, _Inout_ sqlsrv_conn* conn, _Inout_ std::string& conn_str TSRMLS_DC )
+{
+    SQLSRV_ASSERT( Z_TYPE_P( value ) == IS_STRING, "Azure Key Vault keywords accept only strings." );
+        
+    size_t value_len = Z_STRLEN_P( value );
+
+    CHECK_CUSTOM_ERROR( value_len <= 0, conn, SQLSRV_ERROR_KEYSTORE_INVALID_VALUE ) {
+        throw core::CoreException();
+    }
+    
+    switch( option->conn_option_key )
+    {
+        case SQLSRV_CONN_OPTION_KEYSTORE_AUTHENTICATION: 
+        {
+            char *value_str = Z_STRVAL_P( value );        
+            CHECK_CUSTOM_ERROR( stricmp( value_str, "KeyVaultPassword" ) && stricmp( value_str, "KeyVaultClientSecret" ), conn, SQLSRV_ERROR_INVALID_AKV_AUTHENTICATION_OPTION )
+            {
+                throw core::CoreException();
+            }
+            conn->ce_option.akv_auth = value;
+            conn->ce_option.akv_mode = stricmp( value_str, "KeyVaultPassword" ) ? AKVCFG_AUTHMODE_CLIENTKEY : AKVCFG_AUTHMODE_PASSWORD;
+            conn->ce_option.akv_required = true;
```

Comment:
Setting AKV mode here please double check if indeed the user has given you the correct string... you can't just assume the other must be client key mode

Reply:
It's verfied a few lines previously that KeyStoreAuthentication is set to one of its two acceptable values.